packages/zambdas/src/shared/practitioner/helpers.ts

```
@@ -86,3 +87,21 @@ export const getVisitResources = async (
     practitionerRole,
   };
 };
+
+export function getAttendingPractitionerId(encounter: Encounter): string {
+  const practitionerId = encounter.participant
+    ?.find(
+      (participant) =>
+        participant.type?.find(
+          (type) =>
+            type.coding?.some(
+              (c) =>
+                c.system === PRACTITIONER_CODINGS.Attender[0].system && c.code === PRACTITIONER_CODINGS.Attender[0].code
+            )
+        )
+    )
+    ?.individual?.reference?.replace('Practitioner/', '');
+
+  if (!practitionerId) throw Error('Attending practitioner not found');
```

Does it feel a little aggressive to throw an error here? maybe it should return string or undefined and what ever is calling it should throw the error if its undefined?

---

Agreed, will fix in a separate PR so we can ship this faster

---

Fixing in #3271